### ISSUE TITLE
Notifications support

### ISSUE THREAD

**Issue by keunes** [jkbecker/recipe-monica#1]
Hello,

I [requested in the Ferdi repository](https://github.com/getferdi/recipes/issues/17) to add Monica as a service, and the developers referred to this recipe with the note that it doesn't support notifications. It would be great if that could be added - though I'm not actually sure if Monica supports this yet (given https://github.com/monicahq/monica/issues/2303).

Curious to hear your response!

**Comment by jkbecker** [comment]
Hey @keunes - that is a good idea and I had thought about it when I initially whipped up the recipe.

It seems like other similar recipes just scrape the website in question and extract whatever is needed to generate a notification counter, so from a *technical* standpoint I don't see an issue.

My first question would be: What should these notifications entail? Birthdays/other upcoming dates for this month? Tasks? I'm open to suggestions - when I looked at it I was not sure what would be the most intuitive entity to display.

All that said, I will probably not have any time to look into this until some time in January; if anyone wants to give it a shot, please feel free to go ahead and play around with it!

Looking at other recipes provides some good examples of how to do web scraping for notification count - for example here: https://github.com/meetfranz/recipe-hangoutschat/blob/master/webview.js

**Comment by keunes** [comment]
> Looking at other recipes provides some good examples of how to do web scraping for notification count - for example here: https://github.com/meetfranz/recipe-hangoutschat/blob/master/webview.js

I can't code (yet) so that file you linked me to is abacadabra ;)

> My first question would be: What should these notifications entail? Birthdays/other upcoming dates for this month? Tasks? I'm open to suggestions - when I looked at it I was not sure what would be the most intuitive entity to display.

However, I can help with this by giving my opinion :) Notifications are new things I need to follow up on, often a message of some sort. So, in this respect, I would expect only 1) manually set reminders and 2) automatic reminders (birthday, keep in touch) to be included. As for the timeframe, I would ideally boil it down to the coming 7 days for example, but I can imagine that being difficult to do here, in which case reminders for the current month sounds like a sensible thing to do.

**Comment by jkbecker** [comment]
> I can't code (yet) so that file you linked me to is abacadabra ;)

Fair enough. My JavaScript is rusty as well so by all means if anyone else sees this... Feel free to have at it! I'll try to give it a shot in January.

> [...] I would expect only 1) manually set reminders and 2) automatic reminders (birthday, keep in touch) to be included. As for the timeframe, I would ideally boil it down to the coming 7 days for example, but I can imagine that being difficult to do here, in which case reminders for the current month sounds like a sensible thing to do.

That sounds reasonable to me. Whether or not a 7 day timeframe is substantially harder than the monthy thing is hard to anticipate for me at this point, I'll have to look into how the reminders are represented in the DOM (Document Object Model) and therefore how easy it will be to filter for them by date in a flexible way (rather than just counting everything that is shown on the home page, for example). Once I've taken a closer look I can see what's easily possible and then implement whatever timeframe is possible.
